framework/src/node/generator/generator.ts

```diff
@@ -241,6 +243,37 @@ export class Generator {
 				),
 			);
 		});
+
+		this._consensus.events.on(
+			CONSENSUS_EVENT_FINALIZED_HEIGHT_CHANGED,
+			// eslint-disable-next-line @typescript-eslint/no-misused-promises
+			async ({ from, to }: { from: number; to: number }) => {
+				for (const [address, _] of this._keypairs.entries()) {
+					for (let height = from + 1; height < to + 1; height += 1) {
+						const blockHeader = await this._chain.dataAccess.getBlockHeaderByHeight(height);
+						const blsSecretKey = this._endpoint.blsKeys.get(address);
```

Comment:
similar to keyPair, blsKeys should be managed in generator as well. it should be initialized on `load` similar to keypair

Reply:
But `_loadGenerators` is used for debugging purposes unlike `updateForgingStatus` in the endpoint that is for prod. May you elaborate more on your comment?

Reply:
i think it should behave the same as the generation key handling. it is used for testing in the network, and it is a feature we want to support as well